pkg/v1/tkg/test/tkgctl/shared/upgrade.go

```diff
@@ -175,11 +175,11 @@ func E2EUpgradeSpec(context context.Context, inputGetter func() E2EUpgradeSpecIn
 	})
 }
 
-func validateKubernetesVersion(clusterName string, expectedK8sVersion string) { // nolint:unused
+func validateKubernetesVersion(clusterName, expectedK8sVersion, kubeconfigPath string) { // nolint:unused
```

Comment:
question: why do we need to pass in kubeconfig?

Reply:
Cause ```framework.NewClusterProxy()``` uses that and it was passing "" which uses the default kubeconfig and wouldn't work for the kubeconfig created in the temp directory